mongoengine/queryset/base.py

```diff
@@ -423,7 +424,9 @@ def delete(self, write_concern=None, _from_doc_delete=False):
         if call_document_delete:
             cnt = 0
             for doc in queryset:
-                doc.delete(write_concern=write_concern)
+                # How the fuck did this worked before ???
```

Comment:
No need for the swearing but good question, looks like it would have been silently ignored - good catch.